### DISCUSSION TITLE
SDF training options are available from GUI but not from python script

### DISCUSSION THREAD

**Discussion by StillSpeedo** [NVlabs/instant-ngp#770] [Q&A] [answered]
Hello!

I noticed that for some models default **Raystab** mesh SDF mode doesn't work, please see model with trough-holes:

![sdf_raystab](https://user-images.githubusercontent.com/60139001/178443463-08a80a40-6c06-4472-b36e-7f408d108044.png)

Holes are not holes anymore, but with **Watertight** mesh SDF mode they are ok:

![sdf_watertight](https://user-images.githubusercontent.com/60139001/178443658-637b8e76-10ba-4511-a236-0f803b70ece9.png)

However, I didn't found how to set this params using python _run.py_ script. Could you please tell me if it is possible and if so how?

Thanks in advance!

**Comment by Tom94** [comment, marked as answer]
`testbed.sdf.mesh_sdf_mode = ngp.MeshSdfMode.Watertight` should do the trick.

In general, you can check `python_api.cu` for a list of all bindings. Everything available in the GUI should already have one.

Note that if you use the watertight mode on a mesh that has holes (and is thus not actually watertight), the SDF will try to create a paper-thin surface, which in turn can lead to a number of numerical issues. Here be dragons. :)

**Reply by StillSpeedo** [reply]
So fast and clear answer, you are awesome!
